### PR TITLE
Wifi scan sample

### PR DESCRIPTION
**Pull request by D-Triveni**

### CODE REVIEW

File: samples/wifi/scan/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+if(CONFIG_WPA_SUPP)
+set(WPA_SUPP_BASE ${WEST_TOPDIR}/modules/lib/hostap)
+endif()
+
+target_compile_definitions_ifdef(CONFIG_WPA_SUPP_CRYPTO
+	app
+	PRIVATE
+	CONFIG_WEP
+)
```
Comment (krish2718) [inline samples/wifi/scan/CMakeLists.txt]: Remove, this is only for CONFIG_WPA_SUPP

File: samples/wifi/scan/README.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#. |connect_terminal|
+   The output should be similar to the following::
+
+      Scan requested
+      Num  | SSID                             (len) | Chan | RSSI | Sec
+      1    | dlink-7D64                       10    | 1    | 0    | WPA/WPA2
+      2    | abcd_527_24                      11    | 4    | 0    | Open
+      3    | ASUS_RTAX88U11g                  15    | 3    | 0    | WPA/WPA2
+      4    | TP-Link_6BA8                     12    | 9    | 0    | WPA/WPA2
+      Scan request done
```
Comment (krish2718) [inline samples/wifi/scan/README.rst]: This is old snapshot, please update.

File: samples/wifi/scan/README.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+============
+
+This sample uses the following `sdk-nrfxlib`_ library:
+
+* :ref:`nrfxlib:nrf_security`
+
+This sample also uses modules that can be found in the following locations in the |NCS| folder structure:
+
+* ``modules/lib/hostap``
+* ``modules/mbedtls``
```
Comment (krish2718) [inline samples/wifi/scan/README.rst]: None of these are applicable for SCAN.

File: samples/wifi/scan/prj.conf [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+# Networking
+CONFIG_NETWORKING=y
+CONFIG_NET_LOG=y
+CONFIG_NET_IPV6=y
+CONFIG_NET_IPV4=y
+CONFIG_NET_UDP=y
+CONFIG_NET_TCP=y
+CONFIG_NET_DHCPV4=y
+CONFIG_DNS_RESOLVER=y
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf]: We can remove these, but check if driver gets build.

File: samples/wifi/scan/prj.conf [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+CONFIG_HEAP_MEM_POOL_SIZE=153600
```
Comment (krish2718) [inline samples/wifi/scan/prj.conf]: can reduce as its only scan.

File: samples/wifi/scan/src/main.c [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/net_event.h>
+
+#define WIFI_SHELL_MODULE "wifi"
+
+#define WIFI_SHELL_MGMT_EVENTS (NET_EVENT_WIFI_SCAN_RESULT |		\
+				NET_EVENT_WIFI_SCAN_DONE)
+
+LOG_MODULE_REGISTER(scan, CONFIG_LOG_DEFAULT_LEVEL);
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c]: use the sample name in the logger.

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/scan/prj.conf (hunk 4 above)
Comment (D-Triveni) [reply samples/wifi/scan/prj.conf]: @krish2718  Tried by removing these configs, Scan results event is not processing as expected.

File: samples/wifi/scan/prj.conf (hunk 5 above)
Comment (D-Triveni) [reply samples/wifi/scan/prj.conf]: > can reduce as its only scan.

@krish2718 Reduced to 100000, seeing below errors.
```
<inf> wifi_nrf: wifi_nrf_fmac_fw_load: UMAC boot check passed
<err> wifi_nrf: wifi_nrf_fmac_rx_cmd_send: No space for allocating RX buffer
```

File: samples/wifi/scan/src/main.c (hunk 6 above)
Comment (D-Triveni) [reply samples/wifi/scan/src/main.c]: @krish2718  I've written the sample name "scan" in the logger. 
Is there anything that I can modify here?

Comment (krish2718) [reply samples/wifi/scan/src/main.c]: No.

File: samples/wifi/scan/prj.conf (hunk 5 above)
Comment (krish2718) [reply samples/wifi/scan/prj.conf]: okay, lets leave this as is for now, my memory optimization changes will take care of this.

File: samples/wifi/scan/prj.conf (hunk 4 above)
Comment (krish2718) [reply samples/wifi/scan/prj.conf]: We can leave it as is then, but these are not strictly needed, some are needed for net_mgmt but they should be auto-selected.

File: samples/wifi/scan/README.rst (hunk 3 above)
Comment (D-Triveni) [reply samples/wifi/scan/README.rst]: Addressed.

File: samples/wifi/scan/README.rst (hunk 2 above)
Comment (D-Triveni) [reply samples/wifi/scan/README.rst]: Addressed.

File: samples/wifi/scan/CMakeLists.txt (hunk 1 above)
Comment (D-Triveni) [reply samples/wifi/scan/CMakeLists.txt]: Addressed.

File: samples/wifi/scan/prj.conf [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# Debugging
+CONFIG_STACK_SENTINEL=y
+CONFIG_DEBUG_COREDUMP=y
+CONFIG_DEBUG_COREDUMP_BACKEND_LOGGING=y
+CONFIG_DEBUG_COREDUMP_MEMORY_DUMP_MIN=y
+CONFIG_SHELL_CMDS_RESIZE=n
+
+# Logging
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=y
```
Comment (D-Triveni) [inline samples/wifi/scan/prj.conf]: @krish2718  Few scan-results got truncated by using LOG_INF() and also SSID field is updating empty. So, to fix that, added LOG_MODE_MINIMAL config.

Comment (krish2718) [reply samples/wifi/scan/prj.conf]: Ah, we need to increase the LOG buffer size, but as the original wifi_shell, lets use `printk` itself.

File: samples/wifi/scan/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+cmake_minimum_required(VERSION 3.20.0)
+
+find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
+project(nrf_wifi_scan)
+
+set(WIFI_NRF_DRV_BASE ${WEST_TOPDIR}/nrf/drivers/wifi/nrf700x)
```
Comment (krish2718) [inline samples/wifi/scan/CMakeLists.txt]: Even this can be removed

File: samples/wifi/scan/prj.conf (hunk 7 above)
Comment (D-Triveni) [reply samples/wifi/scan/prj.conf]: @krish2718 Replaced logger with printk.

File: samples/wifi/scan/CMakeLists.txt (hunk 8 above)
Comment (D-Triveni) [reply samples/wifi/scan/CMakeLists.txt]: @krish2718  Addressed.

File: samples/wifi/scan/README.rst [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+   The output should be similar to the following::
+
+      <inf> scan: Scan requested
+      Num  | SSID                             (len) | Chan | RSSI | Security | BSSID
+      1    | WiFi1-5.2.3_24G                  15    | 1    | -37  | WPA/WPA2 | 00:0C:43:48:56:67
+      2    | WiFi2-5.2.3_24G                  15    | 1    | -65  | WPA/WPA2 | 02:0C:43:18:56:67
+      3    | ASUS_53U_2G                      11    | 1    | -41  | WPA/WPA2 | 50:EB:F6:17:D4:40
+      4    | NORDIC-INTERNAL                  15    | 1    | -51  | Open     | F0:1D:2D:73:C4:C1
+      5    | tplink_sleeptest                 16    | 1    | -64  | Open     | E8:DE:27:6D:76:A6
+      6    | WiFi3-5.2.3_24G                  15    | 1    | -38  | WPA/WPA2 | 02:0C:43:28:56:67
```
Comment (krish2718) [inline samples/wifi/scan/README.rst]: We should anonymize these, see shell sample for reference

File: samples/wifi/scan/src/main.c [hunk 10]
```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c]: I know this was based on shell sample, but this should really be 2022 + Nordic-5-clause license.

File: samples/wifi/scan/src/main.c [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+	scan_result = 0U;
+
+	net_mgmt_init_event_callback(&wifi_shell_mgmt_cb,
+			wifi_mgmt_event_handler,
+			WIFI_SHELL_MGMT_EVENTS);
+
+	net_mgmt_add_event_callback(&wifi_shell_mgmt_cb);
+
+	/* For now hardcode to 128MHz */
+	nrfx_clock_divider_set(NRF_CLOCK_DOMAIN_HFCLK,
```
Comment (krish2718) [inline samples/wifi/scan/src/main.c:118]: This should have the ifdef CLOCK_FEATURE_HFCLK_DIVIDE_PRESENT

**Review by krish2718 — APPROVED** [review]
Please close and raise external PR, thanks.

**Comment by D-Triveni** [comment]
Internal review is done. So, closing the PR.

**Pull request closed without merging**